components/public-api-server/pkg/apiv1/oidc_test.go

```diff
@@ -96,13 +99,13 @@ func TestOIDCService_CreateClientConfig_FeatureFlagEnabled(t *testing.T) {
 	})
 
 	t.Run("creates oidc client config", func(t *testing.T) {
-		serverMock, client, dbConn := setupOIDCService(t, withOIDCFeatureEnabled)
+		serverMock, client, dbConn, issuer := setupOIDCService(t, withOIDCFeatureEnabled)
```

Comment:
It took me quite a while to wrap my head around the fact that we're using our server, which has our API handlers, as the issuer, for the validation check during create. 

Perhaps it would make sense to start a second server, and return it as part of `setupOIDCService` which is the issuer server. That would also give us the ability to test for the case when the check fails, which we're currently not covering AFAIK.

Reply:
🙈 indeed, this was a shortcut to have a known to be reachable host. 

💯 will touch on this in follow-up to extract it. As mentioned a while ago, the mocked OIDC services should become proper mocks to get full coverage of the handlers of ours.